src/arrayMethodSort.js

```diff
@@ -3,9 +3,58 @@
 /**
  * Implement method Sort
  */
+
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (compareFunction === undefined) {
+      const copyArray = [...this];
+      const sortArray = [];
+
+      for (let j = 0; j < this.length; j++) {
+        let minValue = copyArray[0];
+        let count = 0;
+
+        for (let i = 0; i < copyArray.length; i++) {
+          if (String(minValue) > String(copyArray[i])) {
+            minValue = copyArray[i]; // find the smallest value
+            count = i;
+          }
+        }
+        sortArray[j] = minValue; // add the value
+        // replace the value with the last one in the array
+        copyArray[count] = copyArray[copyArray.length - 1];
+        copyArray.length -= 1; // make the array shorter
+        count = 0; // reset the counter
+      }
+
+      for (let k = 0; k < sortArray.length; k++) {
+        this[k] = sortArray[k]; // change the input array
+      }
+
+      return this;
```

Comment:
this is complicated solution. You need just to compare as strings
Example: 
```
const compareAsStrings = (a, b) => {
  if ({set_condition_here}) {
    return 0;
  };

  if ({set_another_condition_here}) {
    return 1;
  };

  return -1;
};
```

Reply:
Understood. Fixing it.